test/e2e/zero-ex/v4/index.ts

```diff
@@ -0,0 +1,121 @@
+import { fullMigrateAsync, artifacts } from "@0x/contracts-zero-ex";
+import { BigNumberish, BytesLike, Contract, Wallet } from "ethers";
+import { ethers } from "hardhat";
+
+import { SimpleOrder } from "..";
+import { TypedDataDomain } from "../../../../src/ts";
+
+import { withCompatProvider } from "./compat";
+
+export interface Deployment {
+  contract: Contract;
+  domainSeparator: TypedDataDomain;
+}
+
+export async function deployExchange(deployer: Wallet): Promise<Deployment> {
+  const zeroEx = await withCompatProvider((compatProvider) =>
+    fullMigrateAsync(deployer.address, compatProvider, {
+      from: deployer.address,
+      gas: 9e6,
+    }),
+  );
+
+  const contract = new Contract(
+    zeroEx.address,
+    artifacts.INativeOrdersFeature.compilerOutput.abi,
+    deployer,
+  );
+  const { chainId } = await ethers.provider.getNetwork();
+
+  return {
+    contract,
+    domainSeparator: {
+      name: "ZeroEx",
+      version: "1.0.0",
```

Comment:
I thought it was v4?

Reply:
The domain separator in v4 uses `1.0.0` version string. Go figure :man_shrugging:.